Voga/Infrastructure/Design/Components/VogaLabel/vLabel.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  vLabel.swift
+//  Voga
+//
+//  Created by Ariadne Bigheti on 14/10/21.
+//
+
+import UIKit
+
+final class vLabel: UILabel {
+    
+    init(text: String? = "", textStyle: TextStyle) {
+        super.init(frame: .zero)
+        self.text = text
+        configureTextStyle(for: textStyle)
+        self.textColor = .appColor(.black)
+        self.numberOfLines = 0
+        self.adjustsFontForContentSizeCategory = true
+    }
+    
+    func configureSpacing(horizontal: CGFloat, vertical: CGFloat) {
+        guard let labelText = text else {
+            return
+        }
+        
+        let attributedString = NSMutableAttributedString(string: labelText)
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineSpacing = vertical/10
+        attributedString.addAttributes([NSAttributedString.Key.paragraphStyle : paragraphStyle,  NSAttributedString.Key.kern : 1-horizontal], range: NSRange(location: 0, length: attributedString.length))
+
+        self.attributedText = attributedString
+    }
+        
+    func configureTextStyle(for style: TextStyle) {
```

Comment:
Its fine to have this as a function but we could instead have a stored property with a `didSet` command, so the label would configure itself when textStyle changed, like so:
```suggestion
var textStyle: TextStyle {
didSet {
switch textStyle {
case .title1:
  guard let customFont = UIFont(name: "AtkinsonHyperlegible-Bold", size: 28) else{
                    return
                }
                configureSpacing(horizontal: 0.36, vertical: 34)
                self.font = UIFontMetrics(forTextStyle: .title1).scaledFont(for: customFont)
}
}
}
```